coordinator/controllers/shard_controller.go

```diff
@@ -422,7 +411,15 @@ func (s *shardController) electLeader() (string, error) {
 		s.eventListener.LeaderElected(s.shard, newLeader, maps.Keys(followers))
 	}
 
-	s.keepFencingFailedFollowers(followers)
+	go process.DoWithLabels(
+		s.currentElectionCtx,
+		map[string]string{
+			"oxia":  "election-fencing-failed-followers",
+			"shard": fmt.Sprintf("%d", s.shard),
+		}, func() {
+			s.keepFencingFailedFollowers(s.shardMetadata.Term, s.shardMetadata.Ensemble, s.shardMetadata.Leader, followers)
```

Comment:
We are reading the metadata from the go-routine, outside of its mutex. We can make a clone before or acquire the mutex here